### PR TITLE
User Dashboard

### PR DESCRIPTION
**Pull request by EugeneTeu**
- able to login and logout using google auth
- view task 
- view request
- view history

### CODE REVIEW

**Review by leafgecko — APPROVED** [review]

File: frontend/src/App.css [hunk 1]
```diff
@@ -1,31 +1,37 @@
+@import "~antd/dist/antd.css";
 .App {
-  text-align: center;
+	text-align: center;
 }
 
 .App-logo {
-  height: 40vmin;
-  pointer-events: none;
+	height: 40vmin;
```
Comment (pohlinwei) [inline frontend/src/App.css:7]: Not really crucial but it was just something Weiguang pointed out when he review my code. By using `vmin`, the logo might be unexpectedly small if the screen size is reduced drastically.

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,116 +1,68 @@
...
-                            </Switch>
-                        </React.Suspense>
-            
-                    </UserContext.Provider>
-              </AuthContext.Provider>
-        </BrowserRouter>
-      </div>
+	// A wrapper for <Route> that redirects to the login
+	// screen if you're not yet authenticated.
+	const PrivateRoute = ({ Comp, ...rest }) => {
```
Comment (pohlinwei) [inline frontend/src/App.tsx:21]: :O The indentation 😂😂 You don't have to change it though (I mean I understand that this is probably intended to be the way it is)

File: frontend/src/context/auth.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
-import { createContext, useContext } from 'react';
-import { ProfileObject } from './user';
-
+import { createContext, useContext } from "react";
+import firebase from "firebase";
 // These functions will be created in app.tsx
```
Comment (pohlinwei) [inline frontend/src/context/auth.ts:3]: Do you mean `index.tsx`? I can't seem to find the usage of `AuthContext` in `app.tsx`. Or did I overlook? 😅

File: frontend/src/pages/login/login.module.css [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
-
-    text-align: center;
-    box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22);
-    margin-top: 20vh;
+	margin-left: auto;
+	margin-right: auto;
+	margin-top: 15vh;
+	margin-bottom: 15vh;
+	text-align: center;
+	/* box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22);
```
Comment (pohlinwei) [inline frontend/src/pages/login/login.module.css:7]: Will this still be needed?

File: frontend/src/types.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/** Common types */
+export type Task = {
```
Comment (pohlinwei) [inline frontend/src/types.ts:2]: Let me know if you want to know what are the fields for task/offer/req (:

**Review by pohlinwei — APPROVED** [review]

**Pull request merged**